lib/SIL/Utils/BasicBlockUtils.cpp

```diff
@@ -521,3 +521,36 @@ void swift::findJointPostDominatingSet(
     }
   }
 }
+
+//===----------------------------------------------------------------------===//
+//                          checkReachingBlockDominance
+//===----------------------------------------------------------------------===//
+
+#ifndef NDEBUG
+/// Verify that \p sourceBlock dominates \p destBlock given that some path from
+/// \p sourceBlock to \p destBlock exists.
```

Comment:
Is this pre condition needed? If there is no path from sourceBlock to destBlock, the function will return false.